cql3/statements/alter_keyspace_statement.cc

```diff
@@ -81,33 +86,80 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
+                builder.set_new_keyspace_rf_change_data(_name, _attrs->get_replication_map());
+                service::topology_change change{{builder.build()}};
+                auto topo_schema = qp.db().find_schema(db::system_keyspace::NAME, db::system_keyspace::TOPOLOGY);
+                boost::transform(change.mutations, std::back_inserter(muts), [topo_schema] (const canonical_mutation& cm) {
+                    return cm.to_mutation(topo_schema);
+                });
+
+                service::topology_request_tracking_mutation_builder rtbuilder{utils::UUID{this->global_req_id}};
+                rtbuilder.set("done", false)
+                         .set("start_time", db_clock::now());
+                service::topology_change req_change{{rtbuilder.build()}};
+                auto topo_req_schema = qp.db().find_schema(db::system_keyspace::NAME, db::system_keyspace::TOPOLOGY_REQUESTS);
+                boost::transform(req_change.mutations, std::back_inserter(muts), [topo_req_schema] (const canonical_mutation& cm) {
+                    return cm.to_mutation(topo_req_schema);
+                });
+
+                target_type = event::schema_change::target_type::TABLET_KEYSPACE;
+            } else {
+                throw make_exception_future<std::tuple<::shared_ptr<::cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>(
+                        exceptions::invalid_request_exception("Another global topology request ongoing, please retry."));
+            }
+        } else {
+            auto schema_mutations = service::prepare_keyspace_update_announcement(qp.db().real_database(), ks_md_update, ts);
+            muts.insert(muts.begin(), schema_mutations.begin(), schema_mutations.end());
+        }
 
-        auto m = service::prepare_keyspace_update_announcement(qp.db().real_database(), _attrs->as_ks_metadata_update(old_ksm, tm, feat), ts);
-
-        using namespace cql_transport;
         auto ret = ::make_shared<event::schema_change>(
                 event::schema_change::change_type::UPDATED,
-                event::schema_change::target_type::KEYSPACE,
+                target_type,
                 keyspace());
 
-        return make_ready_future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>(std::make_tuple(std::move(ret), std::move(m), std::vector<sstring>()));
+        return make_ready_future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>(std::make_tuple(std::move(ret), std::move(muts), warnings));
     } catch (data_dictionary::no_such_keyspace& e) {
         return make_exception_future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>(exceptions::invalid_request_exception("Unknown keyspace " + _name));
+    } catch (std::exception& e) {
+        throw;
```

Comment:
why is this catch and throw block needed?

Reply:
It's just some leftover, sorry, removed already.